### ISSUE TITLE
CITextField deprecated - waiting for 4.3.1 (will be removed in Django 5.1)

### ISSUE THREAD

**Issue by davmlaw** [SACGF/variantgrid#831]
```
genes.GeneSymbol.symbol: (fields.W907) django.contrib.postgres.fields.CITextField is deprecated. Support for it (except in historical migrations) will be removed in Django 5.1.
	HINT: Use TextField(db_collation="…") with a case-insensitive non-deterministic collation instead.
genes.GeneSymbolAlias.alias: (fields.W907) django.contrib.postgres.fields.CITextField is deprecated. Support for it (except in historical migrations) will be removed in Django 5.1.
	HINT: Use TextField(db_collation="…") with a case-insensitive non-deterministic collation instead.
```

https://docs.djangoproject.com/en/4.2/ref/contrib/postgres/operations/#manage-postgresql-collations

https://adamj.eu/tech/2023/02/23/migrate-django-postgresql-ci-fields-case-insensitive-collation/

This has been in Shariant Test ever since the last deploy to Shariant Prod.
It's another solution to make sure that Gene Symbols are looked up case insensitively, while still knowing what case they are. There's no specific end user way to test this except to make sure Gene Symbol page still works and condition matching (at the gene symbol level) still works, though it would have already come up by now as when this kind of thing fails it does so loudly.

**Comment by davmlaw** [comment]
I followed the steps above, and it failed:

```
django.db.utils.NotSupportedError: nondeterministic collations are not supported for operator class "text_pattern_ops"
```

This has been [reported](https://code.djangoproject.com/ticket/34505) and fixed, but has not been released yet (isn't in 4.2.2)

I'll disable the warnings for now...

**Comment by davmlaw** [comment]
1st we need to work out what version migrations will work on

Here are the actively updating server Django versions

| Server  | Django version |
| ------------- | ------------- |
| variantgrid.com  | 4.2.10  |
| test.variantgrid.com  | 4.2.14  |
| shariant.org.au | |
| test.shariant.org.au | |
| demo.shariant.org.au | |

**Comment by TheMadBug** [comment]
@davmlaw I've filled in the Django versions of the Shariant environments

**Comment by davmlaw** [comment]
> JA: Side note, I wonder if it's worth putting the Django version as an admin only field on the versions page

Yeah I went and looked hoping it was there - spun off into mini issue SACGF/variantgrid#1129

**Comment by davmlaw** [comment]
It is fixed in 4.2 but only for new ones https://code.djangoproject.com/ticket/33901#comment:17

> This fix indeed only solves the problem when you first time creating model fields with collation and prevents pattern ops index created

So - I had to do a few migrations to rename symbol to "old_symbol", make a new one, copy across, delete old, change PK

**Comment by davmlaw** [comment]
testing: should be no visible change - gene names should remain case insensitive

I searched for GaTa2 and brCa1 and ti worked as before

**Comment by davmlaw** [comment]
https://adamj.eu/tech/2023/02/23/migrate-django-postgresql-ci-fields-case-insensitive-collation/#adjust-queries

If you find errors like:

> NotSupportedError: nondeterministic collations are not supported for LIKE

I added ```get_deterministic_queryset``` which annotated ```symbol_deterministic``` which you can perform likes on
